app/src/main/java/com/github/se/signify/ui/screens/profile/SettingsScreen.kt

```diff
@@ -76,165 +84,253 @@ fun SettingsScreen(
         verticalAlignment = Alignment.CenterVertically,
-        horizontalArrangement = Arrangement.SpaceBetween) {
-          val modeText =
-              if (isDarkTheme) stringResource(R.string.dark_mode_text)
-              else stringResource(R.string.light_mode_text)
-          Text(
-              text = modeText,
-              style = MaterialTheme.typography.bodyLarge,
-              color = MaterialTheme.colorScheme.onBackground)
-          Switch(checked = isDarkTheme, onCheckedChange = { onThemeChange(it) })
+        horizontalArrangement = Arrangement.Center) {
+          Icon(
+              imageVector = Icons.Outlined.Edit,
+              contentDescription = "Edit Username",
+              tint = MaterialTheme.colorScheme.onBackground)
+
+          Spacer(modifier = Modifier.width(8.dp))
+
+          TextField(
+              value = newName,
+              onValueChange = { newName = it },
+              placeholder = { Text(userName, color = MaterialTheme.colorScheme.onBackground) },
+              modifier =
+                  Modifier.widthIn(max = 200.dp)
+                      .padding(vertical = 8.dp)
+                      .background(MaterialTheme.colorScheme.background)
+                      .onFocusChanged { focusState ->
+                        if (!focusState.isFocused && newName != userName && newName.isNotEmpty()) {
+                          showConfirmationDialog.value = true
+                        }
+                      }
+                      .testTag("usernameTextField"),
+              colors =
+                  TextFieldDefaults.colors(
+                      focusedContainerColor = MaterialTheme.colorScheme.background,
+                      unfocusedContainerColor = MaterialTheme.colorScheme.background,
+                      focusedTextColor = MaterialTheme.colorScheme.onBackground,
+                      cursorColor = MaterialTheme.colorScheme.onBackground),
+              keyboardOptions = KeyboardOptions.Default.copy(imeAction = ImeAction.Done),
+              keyboardActions =
+                  KeyboardActions(
+                      onDone = {
+                        if (newName != userName && newName.isNotEmpty()) {
+                          showConfirmationDialog.value = true
+                        }
+                      }),
+              singleLine = true)
         }
-    // Switch between English and French
-    LanguageSwitch(isFrench, onLanguageChange)
+  }
+  ConfirmationDialog(
+      showConfirmationDialog,
+      onConfirm = { userViewModel.updateUserName(newName) },
+      onDismiss = { newName = "" },
+      title = stringResource(R.string.confirm_changes_title),
+      message = stringResource(R.string.confirm_changes_message))
+}
+
+@Composable
+fun ConfirmationDialog(
+    showDialog: MutableState<Boolean>,
+    onConfirm: () -> Unit,
+    onDismiss: () -> Unit,
+    title: String,
+    message: String
+) {
+  if (showDialog.value) {
```

Comment:
This if statement feels like it should be moved out of your `@Composable()`. It feels weird that you call `ConfirmationDialog()`s everywhere, but they aren't actually displayed most of the time.